homeassistant/components/palazzetti/config_flow.py

```diff
@@ -0,0 +1,51 @@
+"""Config flow for Palazzetti."""
+
+from typing import Any
+
+from pypalazzetti.client import PalazzettiClient
+from pypalazzetti.exceptions import CommunicationError
+import voluptuous as vol
+
+from homeassistant.config_entries import ConfigFlow, ConfigFlowResult
+from homeassistant.const import CONF_HOST, CONF_MAC, CONF_NAME
+from homeassistant.helpers import device_registry as dr
+
+from .const import DOMAIN, LOGGER
+
+
+class PalazzettiConfigFlow(ConfigFlow, domain=DOMAIN):
+    """Palazzetti config flow."""
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> ConfigFlowResult:
+        """User configuration step."""
+        errors: dict[str, str] = {}
+        if user_input is not None:
+            host = user_input[CONF_HOST]
+            palazzetti = PalazzettiClient(hostname=host)
+            try:
+                await palazzetti.connect()
+            except CommunicationError:
+                LOGGER.exception("Communication error")
+                errors["base"] = "invalid_host"
+            else:
+                formatted_mac = dr.format_mac(palazzetti.mac)
+
+                # Assign a unique ID to the flow
+                await self.async_set_unique_id(formatted_mac)
+
+                # Abort the flow if a config entry with the same unique ID exists
+                self._abort_if_unique_id_configured()
+
+                name = palazzetti.name
```

Comment:
Ah so the name comes from the device? Why do we store that in the config entry instead of fetching it at runtime?

Reply:
No need. Removing.